.github/workflows/cron_jerigon_zero_testing.yml

```diff
@@ -0,0 +1,102 @@
+name: Jerigon Zero Testing
+
+on:
+  # Uncomment when ready to run on a schedule
+  # schedule:
+  #   # Run every Sunday at 12:00 AM (UTC)
+  #   - cron: "0 0 * * 0"
+  push:
+    branches: [develop]
+  pull_request:
+    branches:
+      - "**"
+  workflow_dispatch:
+    branches:
+      - "**"
+
+env:
+  CARGO_TERM_COLOR: always
+  REGISTRY: ghcr.io
+
+jobs:
+  jerigon_zero_testing:
+    name: Jerigon Zero Testing - Integration and Benchmarking
+    runs-on: zero-ci
+    steps:
+      - name: Checkout zk_evm code
+        uses: actions/checkout@v4
+
+      - name: Setup Rust Toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+
+      - name: Set up Rust Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+
+      - name: Build the project
+        run: |
+          RUSTFLAGS='-C target-cpu=native -Zlinker-features=-lld -Copt-level=3' cargo build --release
+          sudo sysctl kernel.perf_event_paranoid=0
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run Erigon Network
+        run: |
+          cd ..
+          tar xf "$(pwd)/zk_evm/test_data/erigon-data.tar.gz" || {
+            echo "Failed to extract erigon-data.tar.gz"; exit 1;
+          }
+          docker pull ghcr.io/0xpolygonzero/erigon:feat-zero
+          docker run -d --name erigon \
+            -p 8545:8545 \
+            -v $(pwd):/data \
+            ghcr.io/0xpolygonzero/erigon:feat-zero \
+            --datadir=/data/erigon/execution-data \
+            --http.api=eth,erigon,engine,web3,net,debug,trace,txpool,admin \
+            --http.vhosts=* --ws --http --http.addr=0.0.0.0 \
+            --http.corsdomain=* --http.port=8545 \
+            --metrics --metrics.addr=0.0.0.0 --metrics.port=9001 \
+            --db.size.limit=3000MB || {
+            echo "Failed to start Erigon"; exit 1;
+          }
+
+      - name: Regression Test with Zero Tracer in Real Mode
+        run: |
+          export ETH_RPC_URL="http://localhost:8545"
+          rm -rf proofs/* circuits/* ./proofs.json test.out verify.out leader.out
+          random_numbers=($(shuf -i 1-500 -n 5))
```

Comment:
@Nashtare  @praetoriansentry currently using random 5 blocks for integration testing, any pref for blocks

Reply:
I don't think it is a good idea to take random 5 blocks. Should be all 500 probably, if we run it once a week.

Reply:
ya its just for testing, we will choose whatever we think is fine. i was thinking somewhere around 100